src/test/java/uk/gov/ons/ctp/response/collection/exercise/service/impl/EventServiceImplTest.java

```diff
@@ -76,13 +105,113 @@ public void givenEventAlreadyExistsWhenEventIsCreatedThenExceptionIsThrown() {
     }
   }
 
-  private static Event createEvent(EventService.Tag tag) {
-    Timestamp eventTime = new Timestamp(new Date().getTime());
+  /**
+   * Test CTP exception thrown if no business case type override found, so no action plans have been
+   * associated to CE
+   */
+  @Test
+  public void testCreateActionRulesRaisesCTPExceptionIfNoBCaseOverRide() {
+    String tag = EventService.Tag.mps.name();
     Event event = new Event();
-    event.setTimestamp(eventTime);
-    event.setTag(tag.name());
+    CollectionExercise collex = new CollectionExercise();
+    CaseTypeOverride biCaseTypeOverride = new CaseTypeOverride();
+    collex.setExercisePK(1);
+    event.setTag(tag);
 
-    return event;
+    when(caseTypeOverrideRepo.findTopByExerciseFKAndSampleUnitTypeFK(1, "B")).thenReturn(null);
+    when(caseTypeOverrideRepo.findTopByExerciseFKAndSampleUnitTypeFK(1, "BI"))
+        .thenReturn(biCaseTypeOverride);
+    try {
+      eventService.createActionRulesForEvent(event, collex);
+      fail("Trying to create action rules when no action plans associated");
+    } catch (CTPException e) {
+      assertEquals(CTPException.Fault.RESOURCE_NOT_FOUND, e.getFault());
+    }
+  }
+
+  /**
+   * Test CTP exception thrown if no business individual case type override found, so no action
+   * plans have been associated to CE
+   */
+  @Test
+  public void testCreateActionRulesRaisesCTPExceptionIfNoBICaseOverRide() {
+    String tag = EventService.Tag.mps.name();
+    Event event = new Event();
+    CollectionExercise collex = new CollectionExercise();
+    CaseTypeOverride bCaseTypeOverride = new CaseTypeOverride();
+    collex.setExercisePK(1);
+    event.setTag(tag);
+
+    when(caseTypeOverrideRepo.findTopByExerciseFKAndSampleUnitTypeFK(1, "B"))
+        .thenReturn(bCaseTypeOverride);
+    when(caseTypeOverrideRepo.findTopByExerciseFKAndSampleUnitTypeFK(1, "BI")).thenReturn(null);
+    try {
+      eventService.createActionRulesForEvent(event, collex);
+      fail("Trying to create action rules when no action plans associated");
+    } catch (CTPException e) {
+      assertEquals(CTPException.Fault.RESOURCE_NOT_FOUND, e.getFault());
+    }
+  }
+
+  @Test
+  public void testCreateCorrectActionRulesForAnyEvent() throws CTPException {
+    // Given
+    String tag = EventService.Tag.mps.name();
```

Comment:
`tag` is only used once, is it worth assigning it to a variable? (this goes for a few instances over the next few tests)